src/VM/Handlers/Opcode8046Handler.cpp

```diff
@@ -35,9 +37,21 @@ Opcode8046Handler::Opcode8046Handler(VM* vm) : OpcodeHandler(vm)
 
 void Opcode8046Handler::_run()
 {
-    Logger::debug("SCRIPT") << "[8046] [*] - value (change sign)" << std::endl;
-    auto value = _vm->popDataInteger();
-    _vm-> pushDataInteger(-value);
+    Logger::debug("SCRIPT") << "[8046] [*] op_negate" << std::endl;
+    auto value = _vm->dataStack()->pop();
+    if (value->type() == VMStackValue::TYPE_INTEGER) 
+    {
+        _vm->pushDataInteger(- (dynamic_cast<VMStackIntValue*>(value))->value());
+    }
+    else
+    if (value->type() == VMStackValue::TYPE_FLOAT) 
+    {
+        _vm->pushDataFloat(- (dynamic_cast<VMStackFloatValue*>(value))->value());
+    }
+    else
+    {
```

Comment:
I think we need additional method to print run-time warnings for scripts, such as this. So basically there will be two types of errors in scripts:
1) Critical error (like non-object passed where object is expected), prints to debug.log, halts current script invocation, doesn't crash the game.
2) Non-critical error, just print to debug.log. But I don't want to use Logger directly, as there might be a lot of those, there needs to be another small abstraction.
This is an idea for another pull-request.